pkg/build/admission/defaults/admission.go

```diff
@@ -130,6 +141,18 @@ func (a *buildDefaults) applyBuildDefaults(build *buildapi.Build) {
 			build.Spec.Source.Git.NoProxy = &t
 		}
 	}
+
+	//apply default source secret if one set after all validation
+	//BUG: build fails because secret is not mounted... where to mount it?
```

Comment:
@Mangirdaz I tried with a build locally and cannot reproduce what you're seeing. In my case, the secret gets mounted in the pod when I specify it in the build.spec.sourceSecret field. What version of the server are you using?